src/VASSAL/build/module/Chatter.java

```diff
@@ -21,477 +21,636 @@
-      @Override
+				scroll.getVerticalScrollBar().setValue(scroll.getVerticalScrollBar().getMaximum());
+			}
+		});
+
+		input = new JTextField(60);
+		input.setFocusTraversalKeysEnabled(false);
+		input.addActionListener(new ActionListener() {
+			@Override
       public void actionPerformed(ActionEvent e) {
-        send(formatChat(e.getActionCommand()));
-        input.setText(""); //$NON-NLS-1$
-      }
-    });
-    input.setMaximumSize(new Dimension(input.getMaximumSize().width,
-                                       input.getPreferredSize().height));
-    scroll.setViewportView(conversation);
-    add(scroll);
-    add(input);
-  }
+				send(formatChat(e.getActionCommand()));
+				input.setText(""); //$NON-NLS-1$
+			}
+		});
+		input.setMaximumSize(new Dimension(input.getMaximumSize().width, input.getPreferredSize().height));
 
-  private String formatChat(String text) {
-    final String id = GlobalOptions.getInstance().getPlayerId();
-    return "<" + (id.length() == 0 ? "("+getAnonymousUserName()+")" : id) + "> - " + text; //$NON-NLS-1$ //$NON-NLS-2$
-  }
+		conversation.setPreferredSize(new Dimension(input.getMaximumSize().width, input.getPreferredSize().height * 11));
 
-  public JTextField getInputField() {
-    return input;
-  }
-
-  /**
-   * Display a message in the text area
-   */
-  public void show(String s) {
-    conversation.append("\n" + s); //$NON-NLS-1$
-  }
-
-  /** @deprecated use GlobalOptions.getPlayerId() */
-  @Deprecated public void setHandle(String s) {
-  }
+		scroll.setViewportView(conversation);
+		scroll.getVerticalScrollBar().setUnitIncrement(input.getPreferredSize().height); // BR// Scroll this faster																									
+		add(scroll);
+		add(input);
+	}
 
-  /** @deprecated use GlobalOptions.getPlayerId() */
-  @Deprecated public String getHandle() {
-    return GlobalOptions.getInstance().getPlayerId();
-  }
+	private String formatChat(String text) {
+    final String id = GlobalOptions.getInstance().getPlayerId();
+    return "&lt;" + (id.length() == 0 ? "(" + getAnonymousUserName() + ")" : id) + "&gt; - " + text; //$NON-NLS-1$ //$NON-NLS-2$ //BR//HTML-friendly angle brackets
+	}
 
-  /**
-   * Set the Font used by the text area
-   */
-  @Override
+	public JTextField getInputField() {
+    return input;
+	}
+	
+	//BR// To make the player colors easy to override in a custom class 
+	//BR// (my modules have logic to assign individual player colors -- beyond the scope of the present effort but a perhaps a fun future addition)
+	protected String getChatStyle(String s)
+	{	  
+    if (s.startsWith(formatChat("").trim())) { //$NON-NLS-1$
+      return "mychat";      
+    } else {
+      return "other";
+    }	  
+	}
+
+	/**
+	 * Display a message in the text area
+	 */
+	public void show(String s) {
+		String style;
+		boolean html_allowed;
+
+		//BR// Choose an appropriate style to display this message in
+		s = s.trim();
+		if (s.length() > 0) {
```

Comment:
A `if (!s.isEmpty())` might be better readable.

Reply:
fixed for next push